Python/pywarpx/picmi.py

```diff
@@ -62,7 +62,9 @@ def init(self, kw):
                 else:
                     self.charge = self.charge_state*constants.q_e
             # Match a string of the format '#nXx', with the '#n' optional isotope number.
-            m = re.match(r'(?P<iso>#[\d+])*(?P<sym>[A-Za-z]+)', self.particle_type)
+            m = None
+            if self.particle_type is not None:
+                m = re.match(r'(?P<iso>#[\d+])*(?P<sym>[A-Za-z]+)', self.particle_type)
             if m is not None:
```

Comment:
Thanks for finding this. Instead of setting `m = None`, can you put the lines 65-80 (old line numbers) inside an if statement `if self.particle_type is not None:`?